fecgov/fecfile-web-api
# [WIP] 651 - Add structlog for json logging in prod

## TODO
- [ ] Check on logging preferences for end-to-end tests
- [ ] Check wtih devs on logging output
- [ ] Review logs to look for opportunities for better data, clean up old-style string formatting to make key/value additions easier
- [ ] Look into exception handler
```python
# TODO: I think this is swallowing exceptions
REST_FRAMEWORK["EXCEPTION_HANDLER"] = "fecfiler.utils.custom_exception_handler"  # noqa
```
- [x] New fields

## Summary

- Issue api#651
  
Add structlog for json logging in prod

## How to test

- Run locally to see new API logs, go to http://localhost:8080/celery-test/ to see celery logs
- To test json, make the below changes, restart celery and go to http://localhost:8080/celery-test/. Check out API and celery logs

**django-backend/fecfiler/celery.py**
```python
def receiver_setup_logging(loglevel, logfile, format, colorize, **kwargs):
...
+    # logging.config.dictConfig(settings.get_env_logging_config(env.space))
+    logging.config.dictConfig(settings.get_env_logging_config(prod=True))
```
**django-backend/fecfiler/settings/base.py**
```python
-# LOGGING = get_env_logging_config()
+LOGGING = get_env_logging_config(prod=True)
```
**django-backend/fecfiler/settings/local.py**
```python
 structlog.configure(  # noqa
-  #  processors=get_env_logging_processors(), # noqa
+    processors=get_env_logging_processors(prod=True), # noqa
```
